doc/subsystems/shell/shell.rst

```diff
@@ -418,6 +418,23 @@ Usage
 *****
 
 Use the :c:macro:`SHELL_DEFINE` macro to create an instance of the shell.
+Pass expected `shell_flag` parameter to this macro.  Otherwise, the shell
+might not move the terminal cursor to a new line correctly.
+
+.. list-table:: Available shell flags
+   :widths: 10 30
+   :header-rows: 1
+
+   * - Flag
+     - Action
+   * - SHELL_FLAG_LFCR_DEFAULT
```

Comment:
Usually this is described as `CRLF` and not `LFCR`. Any reason for the current wording?

Reply:
I do not see a clear rule here. For instance here:
https://linux.die.net/man/1/stty
They are using NLCR or CRNL.

Reply:
Historically, and the preponderance of references say it's ``CRLF``.  That's also the order of the end of line characters on a Windows system:  CR+LF (not LF+CR).  That's also the order in standards docs such as https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4

Is there any reference indicating that the sequence ``LF+CR`` is a valid end-of-line sequence?

What happens if the output string has a ``CR+LF`` sequence in it and you set ``SHELL_FLAG_OLFCR``.  Do you get ``CR+LF+CR``?  Should these flags only add characters if a ``CR+LF`` sequence is not there?

Reply:
Here's an interesting read on the topic too: https://en.wikipedia.org/wiki/Newline